### PR TITLE
Fixed Vivek's bug

### LINKED ISSUE #5850: Confirm Tx Component Crashing & Spinning Forever
I got a report from Vivek at Gitcoin that he was getting "the infinite spinner" (a description that encompasses some number of bugs).

I got enough data from him to reproduce his situation, and it seems that the component [`confirm-transaction.component.js`](https://github.com/MetaMask/metamask-extension/blob/develop/ui/app/components/pages/confirm-transaction/confirm-transaction.component.js) is showing this particular infinite spinner.

This is because his state lacks an `id` associated with the route. I'm not clear where this id should be getting assigned. I noticed one code path in the reducers that assigned the view but not the context, but when I made that also set context it didn't change the result.

Opening this issue to track my investigations as I go through it.

### PR DESCRIPTION
**Pull request by danfinlay**
Fixes #5850

What was happening:

It seems that his MetaMask had crashed while some new transactions had
been loading defaults. He probably had a network connectivity issue to
Infura (which we are working with Infura to address).

As a result of this network cutout, his three unapproved transactions
were not marked failed, and were not marked as `loadingDefaults =
false`, as their gas prices had not yet been estimated.

Normally this behavior is supposed to clean itself up when the
transaction controller starts up, via the
`TransactionController._onBootCleanUp()` function, but in this case,
during unlock, that function was unable to do its job because when it
requested the transaction list, the current network was in the `loading`
state, making it proceed as if there were no pending transactions.

Since we had `unapproved` transactions stuck in the `loadingDefaults` state, this triggered the loading spinner permanently on the confirmation screen.

To fix this, I am doing two things:
- Setting transactions to loadingDefaults = false in more catch blocks.
- Calling `onBootCleanUp()` when the network store's status changes, so
that it will re-trigger when network loading completes.

### CODE REVIEW

**Comment by danfinlay** [comment]
Closing until I fix this bug

**Comment by danfinlay** [comment]
That should do it.

File: app/scripts/controllers/transactions/index.js [hunk 1]
```diff
@@ -485,6 +491,8 @@ class TransactionController extends EventEmitter {
         txMeta.loadingDefaults = false
         this.txStateManager.updateTx(txMeta, 'transactions: gas estimation for tx on boot')
       }).catch((error) => {
+        tx.loadingDefaults = false
+        this.txStateManager.updateTx(tx)
```
Comment (frankiebee) [inline app/scripts/controllers/transactions/index.js]: can you add a not for history in this update so we know that it was set false in the catch and didnt fail for another reason

**Review by frankiebee — DISMISSED** [review]

File: app/scripts/controllers/transactions/index.js (hunk 1 above)
Comment (frankiebee) [reply app/scripts/controllers/transactions/index.js]: actually this is fine

**Review by frankiebee — DISMISSED** [review]
LGTM

File: app/scripts/controllers/transactions/index.js (hunk 1 above)
Comment (danfinlay) [reply app/scripts/controllers/transactions/index.js]: Done.

**Review by frankiebee — APPROVED** [review]
Thank you!

**Pull request merged**